modules/RTC/JitsiLocalTrack.js

```diff
@@ -521,6 +522,7 @@ export default class JitsiLocalTrack extends JitsiTrack {
                         this._unregisterHandlers();
                         this.stopStream();
                         this._setStream(null);
+                        this.emit(TRACK_MUTE_CHANGED, this);
```

Comment:
um moving `this.emit(TRACK_MUTE_CHANGED, this);` from the line 592 doesn't change much only diverges the flow and scatters the update to different places. What were you trying to achieve?

Reply:
I had to move this because we need to wait for the track to be removed/added from the peerconnection before emitting the `TRACK_MUTE_CHANGED` event. Otherwise, the video sender would not be present when I am trying to set the parameters on the sender immediately after the event is fired.

Reply:
do you mean `this._addStreamToConferenceAsUnmute`?

Reply:
I think that's what the old code was doing, it was chaining `this._addStreamToConferenceAsUnmute` on to `promise` and then the events were emitted on line https://github.com/jitsi/lib-jitsi-meet/pull/1119/files/74aff9367bf7343722f0dd4104e966bc3a9529e6#diff-922e8edf325ba2a951da3fe2c72eaad4L592